### DISCUSSION TITLE
NAPALM error when querying an Aruba switch

### DISCUSSION THREAD

**Discussion by Profecy** [netbox-community/netbox#6685] [Q&A] [answered]
Hi everybody,

I set up NAPALM according to the documentation of Netbox and installed the python modules necessary.
I also installed the Aruba NAPALM driver and specified it in my Platform config, the driver name being "aoscx".
Now when I go to a switch device, I see the tabs for NAPALM, but when I click on them, an error display which I can't figure out.

This is the error message:

{"error": "cannot import name 'system'", "exception": "ImportError", "netbox_version": "2.11.7", "python_version": "3.6.8"}

Can anbody provide some insight?

Thank you in advance!

**Comment by candlerb** [comment, marked as answer]
Since the string `import.*system` doesn't occur anywhere in the Netbox Python source, I think it's something to do with napalm or the napalm Aruba driver.  I suggest you try running napalm standalone, and get that working first:

```
cd /opt/netbox
. venv/bin/activate
napalm --vendor VENDOR --user USERNAME --password PASSWORD --optional_args OPTIONAL_ARGS HOSTNAME call get_facts
```

I notice that if I put `--vendor system` then I get a similar error to yours:

```
napalm.base.exceptions.ModuleImportError: Cannot import "system". Is the library installed?
```

But clearly the vendor should be something else like "aoscx".

Note: the correct way to install the napalm modules of interest inside the Netbox venv is to add them to `local_requirements.txt` and run `./upgrade.sh`.  My `local_requirements.txt` contains:

```
napalm
napalm-ros
```

**Reply by Profecy** [reply]
Hi, thanks for the reply :)
I will try your suggestion as soon as I get back to work on monday.
